src/main/java/de/tum/in/www1/artemis/service/connectors/bamboo/BambooService.java

```diff
@@ -705,26 +630,23 @@ private Result addFeedbackToResultNew(Result result, List<BambooBuildResultNotif
 
                 // 1) add feedback for failed test cases
                 for (final var failedTest : job.getFailedTests()) {
-                    String methodName = failedTest.getName(); // in the attribute "methodName", bamboo seems to apply some unwanted logic
-
-                    final String errorMessageString = failedTest.getErrors().stream().map(errorString -> processResultErrorMessage(programmingLanguage, errorString)).reduce("",
-                            String::concat);
-
-                    log.debug("errorMSGString is {}", errorMessageString);
-
-                    createAutomaticFeedback(result, methodName, false, errorMessageString);
+                    result.addFeedback(feedbackService.createFeedbackFromTestCase(failedTest.getName(), failedTest.getErrors(), false, programmingLanguage));
                 }
 
                 // 2) add feedback for passed test cases
                 for (final var successfulTest : job.getSuccessfulTests()) {
-                    String methodName = successfulTest.getName(); // in the attribute "methodName", bamboo seems to apply some unwanted logic
-
-                    createAutomaticFeedback(result, methodName, true, null);
+                    result.addFeedback(feedbackService.createFeedbackFromTestCase(successfulTest.getName(), successfulTest.getErrors(), true, programmingLanguage));
```

Comment:
I would prefer to extract the service call and store it in one variable. It makes it more readable. You did the same a few lines below :)